core/src/main/java/hudson/model/AbstractCIBase.java

```diff
@@ -118,7 +121,10 @@ private void updateComputer(Node n, Map<String,Computer> byNameMap, Set<Computer
             c.setNode(n); // reuse
         } else {
             // we always need Computer for the master as a fallback in case there's no other Computer.
-            if(n.getNumExecutors()>0 || n==Jenkins.getInstance()) {
+            // if the node is not accepting tasks, delay computer creation to allow idle computer die out, can be
+            // turned off by
+            if((n.getNumExecutors()>0 && bypass(FeatureSwitchConfiguration.Feature.DEFER_EXECUTOR_CREATION,n.isAcceptingTasks()))
```

Comment:
if you made bypass a method of the enum then you wouldn't need the `import static`

`FeatureSwitchConfiguration.Feature.DEFER_EXECUTOR_CREATION.isOffOr(n.isAcceptingTasks())`

reads better than `bypass` to me